Mathlib/NumberTheory/FLT/OfOddPrimes.lean

```diff
@@ -0,0 +1,25 @@
+/-
+Copyright (c) 2023 Ruben Van de Velde. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Ruben Van de Velde
+-/
+
+import Mathlib.NumberTheory.FLT.Four
+
+/-!
+# Reduce Fermat's Last Theorem to odd primes
+
+To prove Fermat's Last Theorem, it suffices to prove it for odd prime exponents, and the well-known
+case of exponent 4 (`fermatLastTheoremFour`).
+-/
+
+theorem FermatLastTheoremWith.of_odd_primes
+    (hprimes : ∀ p : ℕ, Nat.Prime p → Odd p → FermatLastTheoremWith ℕ p) : FermatLastTheorem := by
```

Comment:
Is there any way to make `FermatLastTheorem` be Fermat's last theorem and `FermatLastTheorem p` be Fermat's last theorem with exponent p? It feels a bit silly to have to write `FermatLastTheoremWith ℕ p` when the N, Z, and Q versions are all equivalent.

Reply:
Here's the best I can come up with: https://github.com/leanprover-community/mathlib4/pull/7494
Not sure whether it's the best approach though